cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/RuntimeClientService.java

```diff
@@ -443,4 +500,147 @@ private long findProgramFinishTime(List<Message> messages) {
       return -1L;
     }
   }
+
+  /**
+   * An aggregator for messages.
+   */
+  private interface MessageAggregator {
+    Iterator<Message> aggregate(Iterator<Message> input);
```

Comment:
What's the contract of this method?

Reply:
Added a javadoc. It takes an iterator as the input and gives an iterator as the output after aggregation. I wanted to use all the code of TopicRelayers while just adding a new object to aggregate messages. Storing an object for metrics aggregation wouldn't look good for non metrics topic relayers, so I created this interface that aggregates any messages and have the metrics message aggregator implement it.